dotnet/src/Connectors/Connectors.Memory.SqlServer/SqlServerMemoryBuilderExtensions.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.SemanticKernel.Memory;
+
+namespace Microsoft.SemanticKernel.Connectors.SqlServer;
+
+/// <summary>
+/// Provides extension methods for the <see cref="MemoryBuilder"/> class to configure SQL Server or Azure SQL connector.
+/// </summary>
+public static class SqlServerMemoryBuilderExtensions
+{
+    /// <summary>
+    /// Registers SQL Server or Azure SQL connector.
+    /// </summary>
+    /// <param name="builder">The <see cref="MemoryBuilder"/> instance.</param>
+    /// <param name="connectionString">Database connection string.</param>
+    /// <param name="schema">Schema of collection tables.</param>
+    /// <returns>Updated Memory builder including Postgres memory connector.</returns>
+    public static MemoryBuilder WithSqlServerMemoryStore(
```

Comment:
Should discuss the naming here - at least initially, this functionality is only expected to be available in SQL Azure, whereas SqlServer may point more towards the on-premise solution. This also probably covers the name of the package itself (i.e. Connectors.Memory.SqlServer). Note that we have the same ambiguity/naming question in EF itself.

We should get input from the product team on this.

Reply:
@roji  - We should to stick to this as we plan to support SQL Vnext. (Initially yes it is Azure SQL DB , MI (serverless) and SQL in Fabric) but we want to indicate future support in SQL Engine too. 
SqlServer engine is every SQL Server and Azure SQL offering so SqlServer is a good name to use